### DISCUSSION TITLE
Best way to set headers for static assets

### DISCUSSION THREAD

**Discussion by dmwallace** [remix-run/remix#7147] [Q&A] [answered]
What's the best way to set headers for static assets? E.g. setting cache-control for fonts,css,images etc served fromt he /public folder?

**Comment by sergiodxa** [comment, marked as answer]
Do it on the http server, if you’re using Remix App Server you will need to switch to Express or you can setup a proxy on production to add the headers.

**Reply by alexander-schranz** [reply]
I have the same problem, I'm fine doing it on the webserver but my remix app by default adds:

```
Cache-Control: public, max-age=3600
```

For non builded files in my public directory. Not sure where this comes from. And so ending up in 2 cache-control headers instead of one.

Do you have any idea @sergiodxa or others?

**Reply by sergiodxa** [reply]
That's being set by the Express server that comes with Remix App Server (remix-serve command), if you want to customize it you need to change from using remix-serve to using a custom HTTP server.

**Reply by alexander-schranz** [reply]
@sergiodxa thx for the response do you have a link to a documentation how to use custom http server or change express defaults headers?

**Reply by sergiodxa** [reply]
You can use the [Express template](https://github.com/remix-run/remix/tree/v2/templates/express) as example
